SemDiff.Core/GitHub.cs

```diff
@@ -127,10 +169,30 @@ public Task UpdateLimitAsync()
             {
                 return null;
             }
+            if (currentSaved != null)
+            {
+                var removePRs = currentSaved;     
+                foreach(var pr in pullRequests)
+                {
+                    foreach(var prRemove in removePRs)
+                    {
+                        if (pr.Number == prRemove.Number)
+                        {
+                            removePRs.Remove(prRemove);
+                            break;
+                        }
+                    }
+                }
+                removeUnusedPRs(removePRs);
+            }
+            currentSaved = pullRequests;
+            
             return await Task.WhenAll(pullRequests.Select(async pr =>
             {
+                pr.LastWrite = DateTime.Now;
                 var files = await HttpGetAsync<IList<Files>>($"/repos/{RepoOwner}/{RepoName}/pulls/{pr.Number}/files");
                 pr.Files = files;
+                UpdateLocalSavedList();
```

Comment:
This actually can't go here, the reason is that all these files are going to be downloaded concurrently. It would be a bad idea to do file IO here because we have no lock around the file.

I would just call it once after the files are downloaded:

```
var pullsToRet = await Task.WhenAll(....);
UpdateLocalSavedList();
return pullsToRet'
```